src/basic/process-util.c

```diff
@@ -2107,7 +2107,7 @@ int posix_spawn_wrapper(
 
                 return FLAGS_SET(flags, POSIX_SPAWN_SETCGROUP);
         }
-        if (ERRNO_IS_NOT_SUPPORTED(r)) {
+        if (ERRNO_IS_NOT_SUPPORTED(r) || r == E2BIG) {
```

Comment:
If this is only an issue due to POSIX_SPAWN_SETCGROUP, then instead of falling back to posix_spawn, please add another branch just above the r == 0 that retries removes POSIX_SPAWN_SETCGROUP from flags and retries pidfd_spawn

Reply:
Thanks, pushed a new version with this change.